src/box/alter.cc

```diff
@@ -1079,23 +1079,8 @@ class ModifySpace: public AlterSpaceOp
 void
 ModifySpace::alter_def(struct alter_space *alter)
 {
-	/*
-	 * Unless it's online space upgrade, Use the old dictionary for the new
-	 * space, because it's already referenced by existing tuple formats.
-	 * We will update it in place in ModifySpace::alter.
-	 *
-	 * For online space upgrade, all tuples fetched from the space will be
-	 * upgraded to the new format before returning to the user so it isn't
-	 * necessary. Moreover, the old tuples may be incompatible with the new
-	 * format so using the new dictionary for them would be wrong and could
-	 * result in error accessing fields by name from the space upgrade
-	 * function.
-	 */
-	if (new_def->opts.upgrade_def == NULL) {
+	if (new_def->opts.upgrade_def == NULL)
 		new_dict = new_def->dict;
```

Comment:
I think we can drop the `new_dict` member, because we can check if `upgrade_def` is `NULL` right in `alter` and `rollback` methods.

Reply:
Done